src/Transaction.ts

```diff
@@ -14,7 +14,8 @@ export class Transaction {
                      contractAddress: string,
                      contractName: string,
                      functionName: string,
-                     parameters: any[]): Promise<any> {
+                     parameters: any[],
+                     estimatedGas?: number): Promise<any> {
     return this.getTxInfos(userAddress)
       .then(txinfos => {
         const txOptions = {
```

Comment:
You need to return the ethFees according to the gasLimit you set. THis is the only reliable thing and determines the max eth spend. 
So ( in pseudocode):
```
gasLimit = estimatedGas ? estimatedGas * 1.2 : 2000000
txOptions.gasLimit = gasLimit
...
...
ethFees: this.utils.formatAmount(
            gasLimit * txOptions.gasPrice, 18
          )
```